Ember-cli install fails with build errors

Versions:
  windows 7
  node 0.12.2
  npm 2.7.4

I am new to node and npm, just trying to get started with ember, and I am stuck at the very first step.   Plz help, what am I doing wrong? "npm install -g ember-cli" has the following results:

> ws@0.5.0 install C:\Users\mjeghers\AppData\Roaming\npm\node_modules\ember-cli\node_modules\testem\node_modules\socket.io\node_modules\engine.io\node_modules\ws
> (node-gyp rebuild 2> builderror.log) || (exit 0)

[edit] full stack -> https://gist.github.com/stefanpenner/f8fdabb80328b7c07c85

Here is the errors occurring with "ember new" (perhaps this should be filed as a new and separate issue?)...

C:\Users\mjeghers>cd C:\Users\mjeghers\Desktop\UUI\ember-tutorial

C:\Users\mjeghers\Desktop\UUI\ember-tutorial>ember -v
version: 0.2.3
node: 0.12.2
npm: 2.7.6

C:\Users\mjeghers\Desktop\UUI\ember-tutorial>ember new uishell
version: 0.2.3
installing
  create .bowerrc
  create .editorconfig
  create .ember-cli
  create .jshintrc
  create .travis.yml
  create Brocfile.js
  create README.md
  create app\app.js
  create app\components\.gitkeep
  create app\controllers\.gitkeep
  create app\helpers\.gitkeep
  create app\index.html
  create app\models\.gitkeep
  create app\router.js
  create app\routes\.gitkeep
  create app\styles\app.css
  create app\templates\application.hbs
  create app\templates\components\.gitkeep
  create app\views\.gitkeep
  create bower.json
  create config\environment.js
  create .gitignore
  create package.json
  create public\crossdomain.xml
  create public\robots.txt
  create testem.json
  create tests\.jshintrc
  create tests\helpers\resolver.js
  create tests\helpers\start-app.js
  create tests\index.html
  create tests\test-helper.js
  create tests\unit\.gitkeep
  create vendor\.gitkeep
Installing packages for tooling via npm.
C:\Users\mjeghers\Desktop\UUI\ember-tutorial\uishell\node_modules\ember-cli\node
_modules\testem\node_modules\socket.io\node_modules\engine.io\node_modules\ws>if
 not defined npm_config_node_gyp (node "C:\Users\mjeghers\AppData\Roaming\npm\no
de_modules\ember-cli\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-
gyp\bin\node-gyp.js" rebuild )  else (rebuild)
Installing packages for tooling via npm..Building the projects in this solution
one at a time. To enable parallel build, please add the "/m" switch.
Installing packages for tooling via npm.  bufferutil.cc
Installing packages for tooling via npm..\src\bufferutil.cc(32): warning C4996:
'NanSymbol': was declared deprecated [
C:\Users\mjeghers\Desktop\UUI\ember-tutorial\uishell\node_modules\ember-cli\nod
e_modules\testem\node_modules\socket.io\node_modules\engine.io\node_modules\ws\
build\bufferutil.vcxproj]
          C:\Users\mjeghers\Desktop\UUI\ember-tutorial\uishell\node_modules\emb
  er-cli\node_modules\testem\node_modules\socket.io\node_modules\engine.io\node
  _modules\ws\node_modules\nan\nan.h(616) : see declaration of 'NanSymbol'
Installing packages for tooling via npm.     Creating library C:\Users\mjeghers\
Desktop\UUI\ember-tutorial\uishell\node
  _modules\ember-cli\node_modules\testem\node_modules\socket.io\node_modules\en
  gine.io\node_modules\ws\build\Release\bufferutil.lib and object C:\Users\mjeg
  hers\Desktop\UUI\ember-tutorial\uishell\node_modules\ember-cli\node_modules\t
  estem\node_modules\socket.io\node_modules\engine.io\node_modules\ws\build\Rel
  ease\bufferutil.exp
Installing packages for tooling via npm..  Generating code
Installing packages for tooling via npm...  Finished generating code
Installing packages for tooling via npm.  bufferutil.vcxproj -> C:\Users\mjegher
s\Desktop\UUI\ember-tutorial\uishell\no
  de_modules\ember-cli\node_modules\testem\node_modules\socket.io\node_modules\
  engine.io\node_modules\ws\build\Release\\bufferutil.node
Installing packages for tooling via npm..  validation.cc
Installing packages for tooling via npm..\src\validation.cc(112): warning C4996:
 'NanSymbol': was declared deprecated
[C:\Users\mjeghers\Desktop\UUI\ember-tutorial\uishell\node_modules\ember-cli\no
de_modules\testem\node_modules\socket.io\node_modules\engine.io\node_modules\ws
\build\validation.vcxproj]
          C:\Users\mjeghers\Desktop\UUI\ember-tutorial\uishell\node_modules\emb
  er-cli\node_modules\testem\node_modules\socket.io\node_modules\engine.io\node
  _modules\ws\node_modules\nan\nan.h(616) : see declaration of 'NanSymbol'
     Creating library C:\Users\mjeghers\Desktop\UUI\ember-tutorial\uishell\node
  _modules\ember-cli\node_modules\testem\node_modules\socket.io\node_modules\en
  gine.io\node_modules\ws\build\Release\validation.lib and object C:\Users\mjeg
  hers\Desktop\UUI\ember-tutorial\uishell\node_modules\ember-cli\node_modules\t
  estem\node_modules\socket.io\node_modules\engine.io\node_modules\ws\build\Rel
  ease\validation.exp
  Generating code
Installing packages for tooling via npm.  Finished generating code
Installing packages for tooling via npm..  validation.vcxproj -> C:\Users\mjeghe
rs\Desktop\UUI\ember-tutorial\uishell\no
  de_modules\ember-cli\node_modules\testem\node_modules\socket.io\node_modules\
  engine.io\node_modules\ws\build\Release\\validation.node
Installing packages for tooling via npm..
C:\Users\mjeghers\Desktop\UUI\ember-tutorial\uishell\node_modules\ember-cli\node
_modules\testem\node_modules\socket.io\node_modules\socket.io-client\node_module
s\engine.io-client\node_modules\ws>if not defined npm_config_node_gyp (node "C:\
Users\mjeghers\AppData\Roaming\npm\node_modules\ember-cli\node_modules\npm\bin\n
ode-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (rebui
ld)
Installing packages for tooling via npmBuilding the projects in this solution on
e at a time. To enable parallel build, please add the "/m" switch.
Installing packages for tooling via npmC:\Program Files (x86)\MSBuild\Microsoft.
Cpp\v4.0\V120\Microsoft.CppBuild.targe
ts(308,5): error MSB3491: Could not write lines to file "Release\obj\bufferutil
\bufferutil.tlog\bufferutil.lastbuildstate". The specified path, file name, or
both are too long. The fully qualified file name must be less than 260 characte
rs, and the directory name must be less than 248 characters. [C:\Users\mjeghers
\Desktop\UUI\ember-tutorial\uishell\node_modules\ember-cli\node_modules\testem\
node_modules\socket.io\node_modules\socket.io-client\node_modules\engine.io-cli
ent\node_modules\ws\build\bufferutil.vcxproj]
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V120\Microsoft.CppBuild.targe
ts(308,5): error MSB3491: Could not write lines to file "Release\obj\validation
\validation.tlog\validation.lastbuildstate". The specified path, file name, or
both are too long. The fully qualified file name must be less than 260 characte
rs, and the directory name must be less than 248 characters. [C:\Users\mjeghers
\Desktop\UUI\ember-tutorial\uishell\node_modules\ember-cli\node_modules\testem\
node_modules\socket.io\node_modules\socket.io-client\node_modules\engine.io-cli
ent\node_modules\ws\build\validation.vcxproj]
Installed packages for tooling via npm.
Installed browser packages via Bower.
Successfully initialized git.

C:\Users\mjeghers\Desktop\UUI\ember-tutorial>

--- a/bin/prepare-release
+++ b/bin/prepare-release
@@ -3,7 +3,7 @@
 rm -rf node_modules
 rm -rf ./ember-cli-*.tgz
 npm cache clear
-npm i
+npm i --no-optional
 npm link
 npm pack
 npm uninstall -g ember-cli

